ctapipe/irf/irf_classes.py

```diff
@@ -0,0 +1,389 @@
+"""
+Define a parent IrfTool class to hold all the options
+"""
+import astropy.units as u
+import numpy as np
+from astropy.table import QTable
+from pyirf.binning import create_bins_per_decade
+
+from ..core import Component, QualityQuery, traits
+from ..core.traits import Bool, Float, Integer, List, Unicode
+
+
+class ToolConfig(Component):
+
+    gamma_file = traits.Path(
+        default_value=None, directory_ok=False, help="Gamma input filename and path"
+    ).tag(config=True)
+    gamma_sim_spectrum = traits.Unicode(
+        default_value="CRAB_HEGRA",
+        help="Name of the pyrif spectra used for the simulated gamma spectrum",
+    ).tag(config=True)
+    proton_file = traits.Path(
+        default_value=None, directory_ok=False, help="Proton input filename and path"
+    ).tag(config=True)
+    proton_sim_spectrum = traits.Unicode(
+        default_value="IRFDOC_PROTON_SPECTRUM",
+        help="Name of the pyrif spectra used for the simulated proton spectrum",
+    ).tag(config=True)
+    electron_file = traits.Path(
+        default_value=None, directory_ok=False, help="Electron input filename and path"
+    ).tag(config=True)
+    electron_sim_spectrum = traits.Unicode(
+        default_value="IRFDOC_ELECTRON_SPECTRUM",
+        help="Name of the pyrif spectra used for the simulated electron spectrum",
+    ).tag(config=True)
+
+    chunk_size = Integer(
+        default_value=100000,
+        allow_none=True,
+        help="How many subarray events to load at once for making predictions.",
+    ).tag(config=True)
+
+    output_path = traits.Path(
+        default_value="./IRF.fits.gz",
+        allow_none=False,
+        directory_ok=False,
+        help="Output file",
+    ).tag(config=True)
+
+    overwrite = Bool(
+        False,
+        help="Overwrite the output file if it exists",
+    ).tag(config=True)
+
+    obs_time = Float(default_value=50.0, help="Observation time").tag(config=True)
+    obs_time_unit = Unicode(
+        default_value="hour",
+        help="Unit used to specify observation time as an astropy unit string.",
+    ).tag(config=True)
+
+    alpha = Float(
+        default_value=0.2, help="Ratio between size of on and off regions"
+    ).tag(config=True)
+    ON_radius = Float(default_value=1.0, help="Radius of ON region in degrees").tag(
+        config=True
+    )
+
+    max_bg_radius = Float(
+        default_value=5.0, help="Radius used to calculate background rate in degrees"
+    ).tag(config=True)
+
+    max_gh_cut_efficiency = Float(
+        default_value=0.8, help="Maximum gamma purity requested"
```

Comment:
It's not "purity", it's "efficiency" or "recall". I.e. the ratio of surviving true gammas.

Reply:
I'll go with efficiency then.